oshmem/mca/sshmem/ucx/sshmem_ucx_module.c

```diff
@@ -120,11 +113,13 @@ segment_create_internal(map_segment_t *ds_buf, void *address, size_t size,
 
     mem_map_params.field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
                                 UCP_MEM_MAP_PARAM_FIELD_LENGTH |
-                                UCP_MEM_MAP_PARAM_FIELD_FLAGS;
+                                UCP_MEM_MAP_PARAM_FIELD_FLAGS  |
+                                UCP_MEM_MAP_PARAM_FIELD_MEMORY_TYPE;
 
-    mem_map_params.address    = address;
-    mem_map_params.length     = size;
-    mem_map_params.flags      = flags;
+    mem_map_params.address     = address;
+    mem_map_params.length      = size;
+    mem_map_params.flags       = flags;
+    mem_map_params.memory_type = mem_type;
```

Comment:
when failed to allocate, do:
`SSHMEM_VERBOSE(err_level, "ucp_mem_map(memory_type=%s) failed: %s\n", ucs_memory_type_names[mem_type], ucs_status_string(status));`